modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java

```diff
@@ -0,0 +1,760 @@
+    private HttpResponse sendLogoutRequestToPrimaryIS() throws IOException {
+
+        HttpResponse response = sendGetRequest(client, SAML_LOGOUT_URL);
+        EntityUtils.consume(response.getEntity());
+        Assert.assertNotNull(response);
+        return response;
+    }
+
+    private String doLogoutConsentApprovalInSecondaryIS() throws Exception {
+
+        HttpResponse response = sendLogoutApprovalPostWithConsent(client);
+        Assert.assertNotNull(response, "Approval request failed.");
+
+        Header locationHeader = response.getFirstHeader(OAuth2Constant.HTTP_RESPONSE_HEADER_LOCATION);
+        Assert.assertNotNull(locationHeader, "Approval request failed for.");
+        EntityUtils.consume(response.getEntity());
+
+        String logoutResponseToPrimaryIS = locationHeader.getValue();
+
+        response = sendGetRequest(client, logoutResponseToPrimaryIS);
+        return extractValueFromResponse(response, "SAMLResponse", 5);
+    }
+
+    private String doLogoutConsentDenyInSecondaryIS() throws Exception {
+
+        HttpResponse response = sendLogoutDenyPostWithConsent(client);
+        Assert.assertNotNull(response, "Approval request failed.");
+
+        Header locationHeader = response.getFirstHeader(OAuth2Constant.HTTP_RESPONSE_HEADER_LOCATION);
+        Assert.assertNotNull(locationHeader, "Approval request failed for.");
+        EntityUtils.consume(response.getEntity());
+
+        return locationHeader.getValue();
+    }
+
+    private boolean validateLogoutPageContent(String logoutPageContent) {
+
+        return logoutPageContent.contains("location.href = \"index.jsp\"");
+    }
+
+    /**
+     * Send post request with parameters
+     *
+     * @param client
+     * @param urlParameters
+     * @param url
+     * @return
+     * @throws ClientProtocolException
+     * @throws java.io.IOException
+     */
+    public HttpResponse sendPostRequestWithParameters(HttpClient client, List<NameValuePair> urlParameters, String url)
+            throws ClientProtocolException, IOException {
+
+        HttpPost request = new HttpPost(url);
+        request.setHeader("User-Agent", OAuth2Constant.USER_AGENT);
+        request.setEntity(new UrlEncodedFormEntity(urlParameters));
+
+        HttpResponse response = client.execute(request);
+        return response;
+    }
+
+    /**
+     * Send Get request
+     *
+     * @param locationURL - Get url location
+     * @return http response
+     * @throws ClientProtocolException
+     * @throws java.io.IOException
+     */
+    public HttpResponse sendGetRequest(HttpClient client, String locationURL) throws ClientProtocolException,
+            IOException {
+
+        HttpGet getRequest = new HttpGet(locationURL);
+        getRequest.addHeader("User-Agent", OAuth2Constant.USER_AGENT);
+        HttpResponse response = client.execute(getRequest);
+
+        return response;
+    }
+
+    public HttpResponse sendLogoutApprovalPostWithConsent(HttpClient client) throws IOException {
```

Comment:
Shall we make the method descriptions in order to maintain the format. WDYT?

Reply:
Addressed with https://github.com/wso2/product-is/pull/10640/commits/1229a0d9d2cb9bb365a6c0aec4236198bd667a2c.